src/pretix/base/models/discount.py

```diff
@@ -187,6 +197,12 @@ def validate_config(cls, data):
                   'on a minimum value.')
             )
 
+        if data.get('subevent_mode') == cls.SUBEVENT_MODE_DISTINCT and not data.get('benefit_same_products'):
+            raise ValidationError(
+                _('You cannot apply the discount only to bookings of different dates if you are applying the discount '
+                  'to a different set of products than the condition.')
+            )
+
```

Comment:
Should we add an error when `benefit_same_products` is not set, but no products are selected in `benefit_limit_products`?

Reply:
No,

1. validation on many-to-many fields is hard and has pitfalls
2. this is annoying if one deletes the selected product and then naturally arrives in an "invalid" state